Quickstart fails on Centos 7.7

Following the Quckstart instructions in README.md failed for me. I'm using CentOS 7.7 with Python 3.6.8. I think swig needs to be added to the list of prerequisites for Linux. It also doesn't seem to work with swig 2.0.10 - maybe this is expected?

I would supply a patch to the docs, but I'm uncertain exactly what the dependency should be so that it works on all systems. I only have my CentOS box and I'm new to swig.

Without swig installed the tests fail with:

```
     0.00ns CRITICAL cocotb.regression                         regression.py:153  in initialise                      Failed to import module test_endian_swapper_hal: No module named 'hal'
```

It took me longer than it should have to notice that further up it said that it could not find swig:

```
swig -python -outcurrentdir ../hal/endian_swapper_hal.h
make[2]: swig: Command not found
make[2]: *** [endian_swapper_hal_wrap.c] Error 127
make[2]: Leaving directory `/home/user/git/cocotb/examples/endian_swapper/cosim'
make[1]: [hal] Error 2 (ignored)
```

With swig 2.0.10 installed it does better, but still fails:

```
30755.03ns ERROR    cocotb.regression                         regression.py:396  in _score_test                     Test Failed: initial_hal_test (result was SystemError)
                                                                                                                     StopIteration
                                                                                                                     
                                                                                                                     The above exception was the direct cause of the following exception:
                                                                                                                     
                                                                                                                     SystemError: <built-in function delete_endian_swapper_state_t> returned a result with an error set
```

With package swig3 (3.0.12) installed all is well.

Yes, it does. Without swig installed at all:

```
(venv) [user@centos tests]$ make
make results.xml
make[1]: Entering directory `/home/user/git/cocotb/examples/endian_swapper/tests'
mkdir -p sim_build
cd ../cosim && make
make[2]: Entering directory `/home/user/git/cocotb/examples/endian_swapper/cosim'
gcc  -fPIC -pthread -fno-strict-aliasing -O2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 \
-fexceptions --param=ssp-buffer-size=4 -D_GNU_SOURCE \
-fwrapv -DNDEBUG -I/usr/include/python3.6m -c io.c -o io_module.o
gcc -pthread -shared  io_module.o -L/usr/lib64  -o io_module.so
swig -python -outcurrentdir ../hal/endian_swapper_hal.h
make[2]: swig: Command not found
make[2]: *** [endian_swapper_hal_wrap.c] Error 127
make[2]: Leaving directory `/home/user/git/cocotb/examples/endian_swapper/cosim'
make[1]: *** [hal] Error 2
make[1]: Leaving directory `/home/user/git/cocotb/examples/endian_swapper/tests'
make: *** [sim] Error 2
```

Much clearer. I would definitely have spotted that right away. :)

swig 2 vs swig 3 remains a problem, of course, but perhaps I would have known that if I knew more about swig?

Thanks Eric for putting me here.  
If I run this using WSL+Ubuntu both examples work OK.  
If I run them using VBox+Ubuntu (Both 18.04LTS, and with swig3.0.12-1) I get for endian-swapper:
     0.00ns CRITICAL cocotb.regression                         regression.py:153  in initialise                      Failed to import module test_endian_swapper_hal: No module named 'hal'
And for adder:
test -f results.xml
/home/git/cocotb/cocotb/share/makefiles/simulators/Makefile.ghdl:68: recipe for target 'results.xml' failed
make[1]: *** [results.xml] Error 1
make[1]: Leaving directory '/home/git/cocotb/examples/adder/tests'
/home/git/cocotb/cocotb/share/makefiles/Makefile.sim:150: recipe for target 'sim' failed
make: *** [sim] Error 2
That latter did run the simulation however, so I think the issue here is something else.
Hope you can help, and I can be of assistance :-)

--- a/examples/endian_swapper/tests/Makefile
+++ b/examples/endian_swapper/tests/Makefile
@@ -69,7 +69,7 @@ include $(shell cocotb-config --makefiles)/Makefile.sim
 ifeq ($(OS),Linux)
 .PHONY: hal
 hal:
-	-cd ../cosim && make
+	cd ../cosim && make
 
 clean::
 	-cd ../cosim && make clean

